pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +788,35 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
                     check_mat[i][j] = check_mat[i][j-1]
 
     return OneDimensionalArray(seq1._dtype, check_mat[row][col][-1])
+
+def is_sorted(array, start, end):
+    """
+    Checks whether the given array is sorted or not.
+
+    Parameters
+    =========
+    array : OneDimensionalArray
+        The array which is to be checked for sort.
+
+    start : int
+        The starting index of the portion of array to be checked.
+
+    end : int
+        The ending index of the portion of array to be checked.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import OneDimensionalArray, is_sorted
+    >>> arr = OneDimensionalArray(int,[4,3,2,1])
+    >>> is_sorted(arr,0,3)
+    False
+    >>> arr1 = OneDimensionalArray(int,[1,2,3])
+    >>> is_sorted(arr1,0,2)
+    True
+
+    """
+    for i in range(start+1,end+1):
+        if(array[i]<array[i-1]):
```

Comment:
This will not work with `DODA` (the dynamic arrays). An operator is defined in `_comp` (or something similar) in `misc_util.py`. Please use that here. 
Moreover, we should allow a custom comparator (defaulting to `<=`) to check if an array is ordered according to some criteria specified by the passed comparator. The function can also be renamed to `is_ordered`.